source/first-boot.rst

```diff
@@ -1,5 +1,46 @@
 First-Boot Customization
 ========================
 
-This is a placeholder for more documentation about first-boot
-customization.
+First-boot customization is performed using ``cloud-init``.
+``cloud-init`` runs during boot of all Ubuntu instances, discovers the
```

Comment:
"runs during boot of Ubuntu instances" if a datasource is found... (that's post-xenial right now)  While it would be easier to gloss over that it will affect post-xenial images in that first-boot customization must take into account specification of the datasource. :-(